radareorg/cutter
# Add new Analysis section to the Preferences widget

**Your checklist for this pull request**


**Detailed description**

This PR adds a new "Analysis" section to the Preferences dialog, so that the user can modify some of the `anal.*` options and  reanalyze the file without having to do it through the console.

For the sake of completeness, I'm repeating the same considerations I put on #2228 here:

1. To keep it simple, for now I narrowed down the selected options at https://hackmd.io/@Megabeets/H1jBD3Hxv/edit to only those that have boolean values. When I have some feedback on the current code, I can add the remaining options, in this or subsequent PRs.
2. In general, I tried to keep the code as simple as possible and modify existing files as little as possible. For example, I chose to use the `Core()->getConfigb` method directly instead of `Config()->getConfig` to avoid having to modify this one.
3. Similarly, I decided to use the already existing `AnalTask` to handle the analysis. As a consequence, the title of the async task dialog is "Initial Analysis" and you can see the message "Loading the file...", which is not quite accurate:
![screenshot](https://user-images.githubusercontent.com/26287552/88467570-8d790d80-ced8-11ea-98c4-cfaff09035db.png)
However, I thought it was OK, for now at least.
4. The selected options are **not** maintained between two different sessions. However, if you think it's worth it, I can add it, as well as a "Restore defaults" button (just as in the Disassembly section).
5. I chose the icon `cog_light.svg`, since it seemed that a cog is a good representation for "Analysis" and it was already there. Let me know if that's OK, or what other icon would you think fits better. Also, I took into account that for the light theme the color of the icon changes, so added the corresponding image too.

Some other thoughts I had:

6. Now, the analyze button performs anly automated analysis (`aaa`). Maybe in the future we can add other options, or a slider just as in the initial dialog.
7. I did not add any documentation, as I didn't find anything referring to the Preferences dialog, but I can add it if needed.

**Test plan (required)**

1. Open a file without analyzing it
2. Go to Preferences -> Edit -> Analysis. The `anal.jmp.ref` and `anal.jmp.tbl` checkboxes should be the only ones checked.
3. In the console, print the value of some of the `anal.*` options and check they match the checkbox (unchecked-false/checked-true. For instance, `e anal.autoname` should return `false`.
4. Check and uncheck some boxes and verify that the returned value in the console for the corresponding configuration parameter has the correct value.
5. Close the Preferences dialog, reopen it and verify that the Analysis options are the same as before closing it.
6. Click on the analyze button. When the analysis finishes, verify that all the views have refreshed.
7. Change to the light theme and check that the icon of the Analyze section is of the same color than the rest.
![cutter-analysis-dialog](https://user-images.githubusercontent.com/26287552/88467884-4bea6180-cedc-11ea-9f35-cf10d7c999ea.gif)

**Closing issues**
Closes #2228
